project1/src/udpclient.c

```diff
@@ -0,0 +1,82 @@
+// Client side implementation of UDP client-server model
+#include "../include/udp.h"
+
+int sendsrt(char *buf ,int sockfd,struct sockaddr_in servaddr)	//Send to the server.
+{
+	sendto(sockfd, (const char *)buf, strlen(buf), 0, (const struct sockaddr *) &servaddr, sizeof(servaddr));
+  	if(buf[0]=='\n')
+	{
+		printf("close\n");
+		close(sockfd);
+		return 0;
+	}
+	printf("Client -> Server \n");
+	printf("%s\n", buf);
+	return 1;
+}
+
+void receives(char *buf,int sockfd,struct sockaddr_in servaddr)//Receives from the server.
+{
+	int n, len;
+	n = recvfrom(sockfd, (char *)buf, MAXLINE, 0, (struct sockaddr *) &servaddr, &len);
+	buf[n] = '\0';
+	printf("Server -> Client \n");
+	printf("%s\n", buf);
+}
+
+void memsetfunc(struct sockaddr_in addr)
+{
+	memset(&addr, 0, sizeof(addr));
+}
+
+int sock()
+{
+	int sockfd;
+	if ( (sockfd = socket(AF_INET, SOCK_DGRAM, 0)) == -1 )
+	{
+		perror("socket creation failed");
+	}
+	return sockfd;
+}
+
+// Driver code
+int main(char *argv[]) {
+	int sockfd;
+	char buf[MAXLINE];
+	struct sockaddr_in servaddr;
+	
+	// Creating socket file descriptor
+	sockfd = sock();
+	if ( sockfd  == -1 )
+	{
+		printf("server not found\n");
+		return 0;
+	}
+
+	memsetfunc(servaddr);
+
+	// Filling server information
+	servaddr.sin_family = AF_INET;
+	servaddr.sin_port = htons(PORT);
+	servaddr.sin_addr.s_addr = INADDR_ANY;
```

Comment:
lines 59-61 can be joined to memsetfunc to make this function more useful. with the purpose of initiating a sockaddr_in struct.